src/azul/terra.py

```diff
@@ -293,6 +294,10 @@ def insufficient_access(self, resource: str):
         )
 
 
+class EmptySQLResult(Exception):
```

Comment:
```suggestion
class _EmptySQLResult(Exception):
```

and make a member of the class that defines `run_sql`. This will make the usages stand out against those  of the `google.api_core` exceptions.